### DISCUSSION TITLE
How to retrieve an EnumDiscriminant variant from a MyEnum variant?

### DISCUSSION THREAD

**Discussion by seanpianka** [Peternator7/strum#286] [Q&A]
The typenames I'm referencing are from the docs' examples for the trait. I have a `MyEnum` that derives the discriminant traits, and I would like to convert an instance of a `MyEnum` variant into the corresponding variant of `EnumDiscriminants` enum.

Once I achieve that, I'd like to then convert that into a `&str`. My plan is to convert a given `MyEnum` into its discriminant name as a string, and include that as a field in a tracing span field:
```rust
    #[tracing::instrument(
        err,
        skip(self, command),
        fields(command = %command.as_ref::<&str>())
    )]
```
where `command` is an instance of `MyEnum`.

Is this possible?

**Comment by fabianboesiger** [comment]
There is a trait [`IntoDiscriminants`](https://github.com/Peternator7/strum/blob/b023d234dd5113ed82654ee472794ba60af46ff7/strum/src/lib.rs#L222 ) that is implemented when deriving `EnumDiscriminants` which has a `discriminants()` function that returns the respective discriminant.

Note that the latest version on crates.io does not yet support that.
